### ISSUE TITLE
"procedure entry point CreateFile2 could not be located" when running gluon.exe

### ISSUE THREAD

**Issue by Etherian** [gluon-lang/gluon#777]
After pulling the most recent commits on `master` and updating my Rust toolchain to 1.37.0, building the REPL yields a broken executable. When run from PowerShell it immediately exits with code -1073741511, and when run from `cmd.exe` (well, actually [Clink](http://mridgers.github.io/clink)) an error dialog appears, stating 
> The procedure entry point CreateFile2 could not be located in the dynamic link library KERNEL32.dll.

It's odd that the binary would be requesting CreateFile2, since I'm building for and on Windows 7 and [the CreateFile2 function was added in Windows 8](https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-createfile2#requirements).

This behavior is unlikely to be related to Gluon itself but may be caused by a bug in one of its dependencies, so I thought I should mention it. ~~I'll also be opening an issue in the Rust repository in case there is a bug in the toolchain.~~ [was unnecessary]

**Comment by Etherian** [comment]
Commits before 595a0949 (master~7) yield a functioning binary while it and later commits continue to yield a broken one, so the errant behavior does appear to be caused by one of Gluon's dependencies and appears to have been introduced in PR #759.

**Comment by Marwes** [comment]
This is most likely due to https://github.com/kkawakam/rustyline being updated. https://github.com/gluon-lang/gluon/pull/759/commits/ec1d26d30ebf7025e2f43635c86e88c4ef41a5e9 should be the last working commit if that is the case

**Comment by Etherian** [comment]
Yep. ec1d26d3 yields a functional binary; 10ef8cdf does not.

**Comment by Etherian** [comment]
Reverting the RustyLine-related changes made to `repl/src/repl.rs` in 10ef8cdf is enough to build a working binary with the current `master` ba66854b.

That is, the following changes to `repl/src/repl.rs` fix the issue in ba66854b:
```diff
diff --git a/repl/src/repl.rs b/repl/src/repl.rs
index be2dac8a..b15dc443 100644
--- a/repl/src/repl.rs
+++ b/repl/src/repl.rs
@@ -1,6 +1,6 @@
 extern crate gluon_completion as completion;

-use std::{borrow::Cow, error::Error as StdError, path::PathBuf, str::FromStr, sync::Mutex};
+use std::{error::Error as StdError, path::PathBuf, str::FromStr, sync::Mutex};

 use futures::{
     future::{self, Either},
@@ -161,55 +161,13 @@ fn complete(thread: &Thread, name: &str, fileinput: &str, pos: usize) -> GluonRe
         .collect())
 }

-struct Completer {
-    thread: RootedThread,
-    hinter: rustyline::hint::HistoryHinter,
-    highlighter: rustyline::highlight::MatchingBracketHighlighter,
-}
+struct Completer(RootedThread);

 impl rustyline::Helper for Completer {}

-impl rustyline::hint::Hinter for Completer {
-    fn hint(&self, line: &str, pos: usize, ctx: &rustyline::Context) -> Option<String> {
-        self.hinter.hint(line, pos, ctx)
-    }
-}
-
-impl rustyline::highlight::Highlighter for Completer {
-    fn highlight<'l>(&self, line: &'l str, pos: usize) -> Cow<'l, str> {
-        self.highlighter.highlight(line, pos)
-    }
-
-    fn highlight_prompt<'b, 's: 'b, 'p: 'b>(
-        &'s self,
-        prompt: &'p str,
-        default: bool,
-    ) -> Cow<'b, str> {
-        self.highlighter.highlight_prompt(prompt, default)
-    }
-
-    fn highlight_hint<'h>(&self, hint: &'h str) -> Cow<'h, str> {
-        // TODO Detect when windows supports ANSI escapes
-        if cfg!(windows) {
-            Cow::Borrowed(hint)
-        } else {
-            use ansi_term::Style;
-            Cow::Owned(Style::new().dimmed().paint(hint).to_string())
-        }
-    }
-
-    fn highlight_candidate<'c>(
-        &self,
-        candidate: &'c str,
-        completion: rustyline::CompletionType,
-    ) -> Cow<'c, str> {
-        self.highlighter.highlight_candidate(candidate, completion)
-    }
+impl rustyline::hint::Hinter for Completer {}

-    fn highlight_char(&self, line: &str, pos: usize) -> bool {
-        self.highlighter.highlight_char(line, pos)
-    }
-}
+impl rustyline::highlight::Highlighter for Completer {}

 impl rustyline::completion::Completer for Completer {
     type Candidate = String;
@@ -220,7 +178,7 @@ impl rustyline::completion::Completer for Completer {
         pos: usize,
         _: &rustyline::Context,
     ) -> rustyline::Result<(usize, Vec<String>)> {
-        let result = complete(&self.thread, "<repl>", line, pos);
+        let result = complete(&self.0, "<repl>", line, pos);

         // Get the start of the completed identifier
         let ident_start = line[..pos]
@@ -300,11 +258,7 @@ fn new_editor(vm: WithVM<()>) -> IO<Editor> {
     if let Err(err) = history_result {
         warn!("Unable to load history: {}", err);
     }
-    editor.set_helper(Some(Completer {
-        thread: vm.vm.root_thread(),
-        hinter: rustyline::hint::HistoryHinter {},
-        highlighter: rustyline::highlight::MatchingBracketHighlighter::default(),
-    }));
+    editor.set_helper(Some(Completer(vm.vm.root_thread())));
     IO::Value(Editor {
         editor: Mutex::new(editor),
     })
```

**Comment by Marwes** [comment]
Looks like the issue is in ansi_term then.

https://github.com/ogham/rust-ansi-term/commit/23456d5fd32179858a827fd4db4861a624e302a2#diff-0d46108ce4a3c312f023e721b094d40c
